src/c_api/c_api.cc

```diff
@@ -441,6 +443,7 @@ XGB_DLL int XGDMatrixCreateFromMat_omp(const bst_float* data,
   const int nthreadmax = std::max(omp_get_num_procs() / 2 - 1, 1);
   //  const int nthreadmax = omp_get_max_threads();
   if (nthread <= 0) nthread=nthreadmax;
+  int nthread_orig = omp_get_max_threads();
```

Comment:
bug, need to save old nthread (since omp sets the number of threads globally as a state) and restore.  So nthread for DMatrix can be different than for xgboost.